src/platform_impl/macos/event_loop.rs

```diff
@@ -50,6 +55,16 @@ impl<T: 'static> EventLoopWindowTarget<T> {
 
 pub struct EventLoop<T: 'static> {
     window_target: Rc<RootWindowTarget<T>>,
+
+    /// We make sure that the callback closure is dropped during a panic
+    /// by making the event loop own it.
+    ///
+    /// Every other reference should be a Weak reference which is only upgraded
+    /// into a strong reference in order to call the callback but then the
+    /// strong reference should be dropped as soon as possible.
+    callback:
+        Option<Arc<Mutex<Box<dyn FnMut(Event<'_, T>, &RootWindowTarget<T>, &mut ControlFlow)>>>>,
```

Comment:
My assumption was that this has to be `Arc<Mutex>` because the callback is exposed through the global `HANDLER` object. But apparently the borrow checker is still happy if this is just `Rc<RefCell>` which is a complete mystery to me and it makes be a bit worried.

Reply:
I _vaguely_ remember there being something particular about `HANDLER` only being accessed on one thread, but this was over 2 years ago and I frankly don't have a mental model of this anymore.

I optimistically had a reason for putting non-thready things in `EventLoopHandler`, and apparently I also reviewed a PR that furthered that decision: #1108 

I really don't know anymore, though.